requests/models.py

```diff
@@ -414,7 +414,7 @@ def prepare_body(self, data, files, json=None):
         content_type = None
         length = None
 
-        if json is not None:
+        if data == {} and json is not None:
```

Comment:
@kennethreitz why are you skeptical? We should be explicit in what we're expecting to see if only to benefit future "us". `not data` is what was there and caused some minor confusion for @Lukasa when he initially reviewed this.